apycula/tiled_fuzzer.py

```diff
@@ -101,45 +101,174 @@ def recode_idx_gw1nz_1(idx):
         "device": "GW1NS-2C-LQFP144-5",
         "partnumber": "GW1NS-UX2CLQ144C5/I4",
         "recode_idx": recode_idx_gw1ns_2,
+        "has_bottom_quadrants": False,
+        "lw_taps": [1, 0, 3, 2],
     },
     "GW1NS-4": {
         "package": "QFN48",
         "device": "GW1NSR-4C-QFN48-7",
         "partnumber": "GW1NSR-LV4CQN48PC7/I6",
         "recode_idx": recode_idx_gw1ns_4,
+        "has_bottom_quadrants": False,
+        "lw_taps": [2, 1, 0, 3],
     },
     "GW1N-9": {
         "package": "PBGA256",
         "device": "GW1N-9-PBGA256-6",
         "partnumber": "GW1N-LV9PG256C6/I5",
         "recode_idx": recode_idx_gw1n9,
+        "has_bottom_quadrants": True,
+        "lw_taps": [3, 2, 1, 0],
     },
     "GW1N-9C": {
         "package": "UBGA332",
         "device": "GW1N-9C-UBGA332-6",
         "partnumber": "GW1N-LV9UG332C6/I5",
-        "recode_idx": recode_idx_gw1n9, # TODO: recheck
+        "recode_idx": recode_idx_gw1n9,
+        "has_bottom_quadrants": True,
```

Comment:
the clock fuzzer already figures this out without hardcoding values, maybe we can reuse that?

Reply:
Yes, that would be correct. Since the clock fuzzer hasn't worked yet at this point, I'll probably move the whole thing to clock_fuzzer.py.

Reply:
After all, long wires are also a kind of clock.
By the way, a side effect is that there is a free gate from logic to clocks: the input points of the long wires can be used by nextpnr as sources for clock SPINExx

For good or bad, we will see:)

Reply:
Now we use the data from the clock fuzzer! Unfortunately GW1NZ-1 is not friendly and clock 3 is not found:(

Reply:
:(